File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1

        while l <= r:
            mid = (r + l) // 2

            if target == nums[mid]:
                return mid
            
            if nums[l] <= nums[mid]:
                if nums[l] <= target < nums[mid]:
                    r = mid - 1  # Target is in the left sorted half
                else:
                    l = mid + 1  # Target is in the right half
            # Otherwise, the right half must be sorted
            else:
                if nums[mid] < target <= nums[r]:
                    l = mid + 1  # Target is in the right sorted half
                else:
                    r = mid - 1  # Target is in the left half
        
        return -1

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0;
        int r = 0;
        int currSum = 0;
        int minLen = INT_MAX; 

        while (r < static_cast<int>(nums.size())) {
            currSum += nums[r];

            while (currSum >= target) {
                minLen = std::min(minLen, r - l + 1); // Comparing int with int
                currSum -= nums[l];
                l++; // Fixed: Increment left pointer to shrink the window
            }

            r++;
        }

        // Return 0 if no valid subarray was found, otherwise return minLen
        return (minLen == INT_MAX) ? 0 : minLen;
    }
};